NMS-13696: Geographical map documentation (#4351)
OpenNMS/opennms: docs/modules/operation/pages/geographical-map/introduction.adoc

> OpenNMS includes a geographical map which can show nodes with address asset information on a world map, including dynamic alarm status.

Reviewer: This section doesn't have a header so it is not showing up in the table of contents. 
Suggest
= Geographical Maps

> OpenNMS includes a geographical map which can show nodes with address asset information on a world map, including dynamic alarm status.

Reviewer 2: Missing page title for navigation labels
```suggestion
# Geographical Maps

OpenNMS includes a geographical map which can show nodes with address asset information on a world map, including dynamic alarm status.
```

OpenNMS includes a geographical map which can show nodes with address asset information on a world map, including dynamic alarm status.

The geographical map is located in the `Maps` menu or by going to the page of a node that has an address and/or longitude/latitude defined in its asset fields, and following the `View in Geographical Map` link.

> ## Markers and Clusters

Reviewer: ```suggestion
## Markers and clusters
```

> The geographical maps display each node that has an address, or longitude and latitude in the database. When the geographical map is loaded, it will look at each node. If the node has coordinates, it will display it on the map. If the node has an address, it will use a geolocation service to convert the address to coordinates, and then display it on the map.

Reviewer 2: Is this true? I thought that geolocation service handled converting addresses to lat/lon in the background and not when viewing the map.

> The geographical maps display each node that has an address, or longitude and latitude in the database. When the geographical map is loaded, it will look at each node. If the node has coordinates, it will display it on the map. If the node has an address, it will use a geolocation service to convert the address to coordinates, and then display it on the map.

Author: I don't know. Should we ask a developer?

> The geographical maps display each node that has an address, or longitude and latitude in the database. When the geographical map is loaded, it will look at each node. If the node has coordinates, it will display it on the map. If the node has an address, it will use a geolocation service to convert the address to coordinates, and then display it on the map.

Reviewer: Yes, we should ask a developer. I'm not sure of the answer either.

> The geographical maps display each node that has an address, or longitude and latitude in the database. When the geographical map is loaded, it will look at each node. If the node has coordinates, it will display it on the map. If the node has an address, it will use a geolocation service to convert the address to coordinates, and then display it on the map.

Reviewer 3: I'm pretty sure the geomaps only do nodes that have lon/lat, it's up to the backend geo scanner stuff to figure out lon/lat from the address before it gets that far.

> The geographical maps display each node that has an address, or longitude and latitude in the database. When the geographical map is loaded, it will look at each node. If the node has coordinates, it will display it on the map. If the node has an address, it will use a geolocation service to convert the address to coordinates, and then display it on the map.

Reviewer 2: Note: The link should be good, but I didn't render it locally to verify.
```suggestion
The geographical map displays each node that has an associated longitude and latitude in the database. 
If a node does not have coordinates recorded, update the requisition to include address asset fields and verify the xref:admin/geocoder.adoc[geocoder service] is configured and enabled.
```

> The geographical maps will automatically cluster nodes into groups if they are close together. By hovering over a cluster (the circle marker with a number in it), it will show the boundaries of the nodes it contains. Click on a marker to show the summary of the nodes it contains.

Reviewer: What do you mean by "show the boundaries" of the nodes it contains?

> The geographical maps will automatically cluster nodes into groups if they are close together. By hovering over a cluster (the circle marker with a number in it), it will show the boundaries of the nodes it contains. Click on a marker to show the summary of the nodes it contains.

Reviewer 2: Might be "show the contents of the nodes it contains" instead

> The geographical maps will automatically cluster nodes into groups if they are close together. By hovering over a cluster (the circle marker with a number in it), it will show the boundaries of the nodes it contains. Click on a marker to show the summary of the nodes it contains.

Author: https://github.com/OpenNMS/opennms/pull/4351/commits/b4eb075c8c2bdf6a299229afbc83dce6d46e2320

## Status

> A node's marker will display as the color associated with the most severe alarm for that node.  A cluster's marker will display as the color of the highest severity of the nodes the marker contains.

Reviewer: ```suggestion
A node's marker displays as the color associated with the most severe alarm for that node. 
A cluster's marker displays as the color of the highest severity of the nodes the marker contains.
```

> 

Reviewer: Unfortunately, the images are not showing up properly when I build the file locally. I've tried modifying the syntax, but nothing is working. It is possible to put images in tables, so we'll have to figure out what's going wrong.

Author: @Bonrob2 Maybe a stupid question but how do you "build" the docs locally? Do we have a defined procedure for that?
In our company, we add readme files into folders like https://github.com/OpenNMS/opennms/tree/develop/docs with usage/build instructions. Maybe it makes sense to add a readme for this? Otherwise, nobody is able to give it a try, right?

Reviewer: That's a good idea. I'll add a ticket for that.

Author: So how can I give it locally on my computer a try to test, if or why the images are not showing up?
I've read that images in tables can be tricky. Maybe also not worth doing it like this?
As @mmahacek mentioned we could also drop the table and just add the images with titles or similar. Wdyt?

Author: I removed the table and just put the picture and a "title" above.

Click on a marker to show the summary of the nodes it contains.

## Status

A node's marker displays as the color associated with the most severe alarm for that node. 
A cluster's marker displays as the color of the highest severity of the nodes the marker contains.

*Normal*

> image::geographical-map/severity_normal.png[width=25,height=41]

Reviewer: For some reason, these images are not showing up when I build the docs. I've tried troubleshooting, but nothing seems to work.

Reviewer 2: figured it out - antora doesn't like how the name and image:: render.  review notes with a fix incoming.

Reviewer 2: ![image](https://user-images.githubusercontent.com/26419758/165155451-a4471e85-0ff5-46ab-b110-42553518ed61.png)

> | image::geographical-map/green.png[green marker]

Reviewer 2: None of these images are rendering when I try to build the docs

> 
> *Normal*
> image::geographical-map/severity_normal.png[width=25,height=41]
> 
> *Warning*
> image::geographical-map/severity_warning.png[width=25,height=41]
> 
> *Minor*
> image::geographical-map/severity_minor.png[width=25,height=41]
> 
> *Major*
> image::geographical-map/severity_major.png[width=25,height=41]
> 
> *Critical*
> image::geographical-map/severity_critical.png[width=25,height=41]

Reviewer 2: ```suggestion
[options="header, autowidth"]
[cols="1,1a"]
|===
| Severity
| Map Pin

| Normal
| image::geographical-map/severity_normal.png[]

| Warning
| image::geographical-map/severity_warning.png[]

| Minor
| image::geographical-map/severity_minor.png[]

| Major
| image::geographical-map/severity_major.png[]

| Critical
| image::geographical-map/severity_critical.png[]
|===
```
